src/v/storage/segment_utils.h

```diff
@@ -197,10 +198,10 @@ struct clean_segment_value
 };
 
 inline bool is_compactible(const model::record_batch& b) {
-    return !(
-      b.header().type == model::record_batch_type::raft_configuration
-      || b.header().type == model::record_batch_type::archival_metadata
-      || b.header().type == model::record_batch_type::version_fence);
+    static const auto filtered_types = model::offset_translator_batch_types();
+    auto n = std::count(
+      filtered_types.begin(), filtered_types.end(), b.header().type);
+    return n == 0;
```

Comment:
Actually, does this mean previously we'd only keep the latest prefix_truncate batch? Or would all batches be removed? If the former, seems like that would work.

Reply:
was this question addressed @Lazin seems kinda important? cc @andrwng

Reply:
Previously we didn't have `prefix_truncate` batches. We shouldn't touch these batches during compaction. Otherwise the offset translation would break. The tiered-storage needs these batches to be present in the segment after compaction. The read path builds its own offset translator state based on the data inside the segment so if you will remove any batch that participates in offset translation all offsets will be shifted. The prefix truncation stm should have the snapshot with insync offset higher than the offset of the `prefix_truncate` batch at this point (otherwise we wouldn't be running compaction on a segment).